scapy/route6.py

```diff
@@ -265,7 +266,7 @@ def route(self, dst, dev=None):
         # Fill the cache (including dev-specific request)
         k = dst
         if dev is not None:
-            k = dst + "%%" + dev
+            k = dst + "%%" + (dev.pcap_name if not isinstance(dev, basestring) else dev)
```

Comment:
It's dev.pcap_name right ? Cause i wasn't sure if it shouldn't have been dev.name...